modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -209,3 +248,123 @@ def row_apply_func(row):
         isinstance(ptn, new_large_modin_frame._partition_mgr_cls._partition_class)
         for ptn in new_large_modin_frame._partitions.flatten()
     ), "Partitions are not block partitioned after apply."
+
+
+@pytest.mark.skipif(
+    Engine.get() not in ("Dask", "Ray"),
+    reason="Only Dask and Ray engines have virtual partitions.",
+)
+@pytest.mark.parametrize(
+    "axis,virtual_partition_class",
+    ((0, virtual_column_partition_class), (1, virtual_row_partition_class)),
+    ids=["partitions_spanning_all_columns", "partitions_spanning_all_rows"],
+)
+class TestBuildVirtualPartition:
+    def test_from_virtual_partitions_with_call_queues(
+        self, axis, virtual_partition_class
+    ):
+        # reverse the dataframe along the virtual partition axis.
+        if axis == 0:
+
+            def reverse(df):
+                return df.iloc[::-1, :]
+
+        else:
+
+            def reverse(df):
+                return df.iloc[:, ::-1]
+
+        level_zero_blocks_first = [
+            block_partition_class(put(pandas.DataFrame([0]))),
+            block_partition_class(put(pandas.DataFrame([1]))),
+        ]
+        level_one_virtual_first = virtual_partition_class(
+            level_zero_blocks_first, full_axis=False
+        )
+        level_one_virtual_first = level_one_virtual_first.add_to_apply_calls(reverse)
+        level_zero_blocks_second = [
+            block_partition_class(put(pandas.DataFrame([2]))),
+            block_partition_class(put(pandas.DataFrame([3]))),
+        ]
+        level_one_virtual_second = virtual_partition_class(
+            level_zero_blocks_second, full_axis=False
+        )
+        level_one_virtual_second = level_one_virtual_second.add_to_apply_calls(reverse)
+        level_two_virtual = virtual_partition_class(
+            [level_one_virtual_first, level_one_virtual_second], full_axis=True
+        )
+        level_two_virtual_result = level_two_virtual.apply(lambda df: df, num_splits=1)[
+            0
+        ]
+        if axis == 0:
+            expected_df = pandas.DataFrame([1, 0, 3, 2], index=[0, 0, 0, 0])
+        else:
+            expected_df = pandas.DataFrame([[1, 0, 3, 2]], columns=[0, 0, 0, 0])
+        df_equals(
+            level_two_virtual_result.to_pandas(),
+            expected_df,
+        )
+
+    def test_from_block_partition_and_virtual_partition_with_call_queues(
+        self,
+        axis,
+        virtual_partition_class,
+    ):
+        # make function the dataframe along the virtual partition axis.
```

Comment:
Could you clarify this comment here? I'm not sure what "make function the dataframe along the virtual partition axis" means.

Also, could the test be renamed `test_from_block_and_virtual_partition_with_call_queues`?